src/main/scala/nl.knaw.dans.easy.solr4files/components/LdapAuthenticationComponent.scala

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.solr4files.components
+
+import java.net.URI
+import java.util
+import javax.naming.Context._
+import javax.naming._
+import javax.naming.ldap.InitialLdapContext
+
+import nl.knaw.dans.easy.solr4files.{ AuthorisationNotAvailableException, InvalidCredentialsException }
+
+import scala.util.{ Failure, Try }
+
+trait LdapAuthenticationComponent extends AuthenticationComponent {
+  val usersParentEntry: String
+  val securityPrincipal: String
+  val securityCredentials: String
+  val ldapProviderUrl: URI
+  val initialContextFactory: String = "com.sun.jndi.ldap.LdapCtxFactory"
+
+  trait LdapAuthentication extends Authentication {
+
+    def getUser(userName: String, password: String): Try[User] = Try {
+      val ctxt = new InitialLdapContext(ldapEnv(userName, password), null)
```

Comment:
This doesn't work. Suppose I need two stage somehow. First establish a connection with the principal configuration parameters, then verify username and password.